src/Ocelot/Request/Mapper/RequestMapper.cs

```diff
@@ -41,29 +42,35 @@ public async Task<Response<HttpRequestMessage>> Map(HttpRequest request, Downstr
             }
         }
 
-        private static async Task<HttpContent> MapContent(HttpRequest request)
+        private static bool BodyIsSupported(string method)
         {
-            if (request.Body == null || (request.Body.CanSeek && request.Body.Length <= 0))
+            return method != HttpMethod.Get.Method
+                && method != HttpMethod.Head.Method
+                && method != HttpMethod.Trace.Method;
+        }
+
+        private static HttpContent MapContent(string downstreamMethod, HttpRequest request)
+        {
+            if (!BodyIsSupported(downstreamMethod)
+                || !BodyIsSupported(request.Method))
             {

```

Comment:
Should we log an error or warning when the request body not supported

Reply:
It depends on implementation! 😄 
But logging warning is better.
Logging error is wrong because we can still make a request to downstream. Only web server decides what to do with a body. So, in case of web server returned error status we can find the warning in Ocelot's log and tune the route to make the route stable.